projects/ngx-charts-on-fhir/src/lib/data-layer/visible-data.service.ts

```diff
@@ -45,7 +45,10 @@ function byMostRecentValue(a: Dataset, b: Dataset) {
   if (ay == null || by == null) {
     return 0;
   }
-  return by - ay;
+  if (typeof ay === 'number' && typeof by === 'number') {
+    return by - ay;
+  }
+  throw new TypeError('Unsupported y-value types');
```

Comment:
I decided not to bother with string comparisons because we do not currently have any DataLayerServices that return multiple category-scaled datasets in the same layer. If we do add such a service in the future, we will probably need to revisit this code anyways. For now, this code never gets called with string values because there is only 1 medication dataset per layer.